starknet_py/net/client.py

```diff
@@ -228,7 +234,7 @@ async def wait_for_tx(
     # Mutating methods
     async def add_transaction(
         self, tx: Transaction, token: Optional[str] = None
-    ) -> Dict[str, int]:
+    ) -> Dict[str, str]:
```

Comment:
What's returned here?

Reply:
There's no schema exposed in cairo-lang library, gateway's add_transaction that is used in this method just returns `Dict[str, str]`. We could build our own schema but it's part of the ongoing refactor anyway.

Reply:
But do they actually return strings in current version (not 0.9)?

Reply:
I am sure there was at least one endpoint with invalid typing.

Reply:
I took typing straight from cairo-lang 0.8.2.1. It seems like they introduced this change in 0.8.1
[see 0.8.1](https://github.com/starkware-libs/cairo-lang/blob/v0.8.1/src/starkware/starknet/services/api/gateway/gateway_client.py)
[see 0.8.0](https://github.com/starkware-libs/cairo-lang/blob/v0.8.0/src/starkware/starknet/services/api/gateway/gateway_client.py)

Reply:
Ok, just please see if they return strings in response. You can resolve this thread id they do :)

Reply:
I've verified on testnet:

If transaction is successful it will be dict with `code` and `transaction_hash`, both strings.

If transaction fails it returns dict with string `code` and `message` that is also a string (but contains serialized dict, e.g.: `"{'calldata': {0: ['Expected int string, got: \"xyz\".']}}"`).

But I'd say typing is correct since we're not doing any parsing of the output.